src/components/message.js

```diff
@@ -6,12 +6,29 @@
 
 import {Component, createElement, isValidElement} from 'react';
 import PropTypes from 'prop-types';
+import IntlMessageFormat from 'intl-messageformat';
+import memoizeIntlConstructor from 'intl-format-cache';
 import {intlShape, messageDescriptorPropTypes} from '../types';
 import {
   invariantIntlContext,
   shallowEquals,
   shouldIntlComponentUpdate,
 } from '../utils';
+import {formatMessage as baseFormatMessage} from '../format';
+
+const defaultFormatMessage = (descriptor, values) => {
+  if (process.env.NODE_ENV !== 'production') {
+    console.error(
+      '[React Intl] Could not find required `intl` object. <IntlProvider> needs to exist in the component ancestry. Using default message as fallback.'
```

Comment:
Anyone could explain this line code for me? Why throw this error when `process.env.NODE_ENV !== 'production'`?
I got stuck by this error in my project. Thanks a lot!

Reply:
This is a really old version. We no longer support this version. In v5 we throw this all the time